src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -177,13 +181,28 @@ private void post(Work work) {
         try {
             logger.trace("putting work on queue.");
             workQueue.put(work);
+            checkQueueSize();
         } catch (InterruptedException ex) {
             logger.warn("Interrupted while putting work on queue!", ex);
             //TODO check if shutdown
             //TODO try again since it is important
         }
     }
 
+    /**
+     * Checks queue size.
+     */
+    private void checkQueueSize() {
+        int queueSize = workQueue.size();
+        if (queueSize >= WORK_QUEUE_SIZE_WARNING_THRESHOLD) {
+            logger.warn("The Gerrit-trigger incoming events queue contains {} items!"
+                        + " Something might be stuck, or your system can't process the commands fast enough."
+                        + " Try to increase the number of receiving worker threads on the Gerrit configuration page."
```

Comment:
Strange to be talking about configuration pages here. Other consumers are not necessarily a trigger and might not even have a configuration page.

Reply:
To be honest, I just copy-pasted it from https://github.com/sonyxperiadev/gerrit-events/blob/7f3547c6d55946e25e99a847b5160d69e59994ba/src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritSendCommandQueue.java#L174.

Can fix in both (initially I tried to generalise this two classes, but change grew too fast and did not stop).

Reply:
ah my memory bad :) the old warning was only on the out queue I guess, not the in queue .

Reply:
@rsandell should I remove mentioning of "configuration pages" from both?

Reply:
Yes, we can rephrase it a bit I think. Something like:

> The Gerrit incoming events queue contains {} items!
>  Something might be stuck, or your system can't process the commands fast enough.
>  Try to increase the number of receiving worker threads.